src/web/js/check-ui.js

```diff
@@ -297,220 +477,90 @@
-                                  ch:   get(loc, "end-column") },
-                                { inclusiveLeft:false,
-                                  inclusiveRight:false},
-                                function(editor, handle) {
-                                  var gutter =
-                                    editor.setGutterMarker(
-                                      get(loc, "start-line") - 1,
-                                      "test-marker-gutter",
-                                      marker[0]);
-                                  
-                                  handle.on("hide",
-                                            function(){
-                                              marker.hide();
-                                            });
-                                  
-                                  handle.on("unhide",
-                                            function(){
-                                              marker.show();
-                                            });
-                                });
-                            }
-                            reason.replaceWith(dom.addClass("test-reason"));
-                          });
-                        }, function(reasonResult) {
-                          if(test_index === 0) {
-                            block.on("click", ".check-block-header",
-                                     function(e) {
-                                       if(block.hasClass("expanded"))
-                                         wrapper.trigger('click');
-                                     });
-                          }
-                          block.one("click", ".check-block-header", function(e) {
-                            if (currentlyExpandedBlock === block && neverOpened){
-                              outputUI.runMarks();
-                              politely(block.find(".cm-future-snippet"),
-                                       function(snip){snip.cmrefresh();});
-                              neverOpened = false;
-                            }
-                          });
-                          restarter.resume(runtime.nothing);
-                        });
-                      } else {
-                        console.error("ARRRGGH BAD THINGS HAPPENED");
-                        restarter.resume(runtime.nothing);
-                      }
-                    })
-                });
+
+      container.append($(checkResultsContainer).prepend(summary));
+      
+      }catch(e){console.error(e);}
+      
+      let maybeLocToAST   = outputUI.makeMaybeLocToAST(runtime, documents, srcloc);
+      let srclocAvaliable = outputUI.makeSrclocAvaliable(runtime, documents, srcloc);
+      let maybeStackLoc   = runtime.makeFunction(function(n, userFramesOnly) {return runtime.ffi.makeNone();});
+      
+      function vivifySkeletonFallback(restarter, skeleton) { 
```

Comment:
Please mark which of these functions must be run on a valid pyret stack, or must be run while not on one.  Stack safety is subtle, and hard to reconstruct the logic for